File: src/Web/WebAngular/src/app/domain/view-models/pipeline-trigger-event.view-model.ts
import { PipelineTriggerFilterViewModel } from "./pipeline-trigger-filter.view-model";

export interface PipelineTriggerEventViewModel {
	id: string;
	pipelineTriggerId: string;
	triggerEventId: string;
	pipelineTriggerFilters: PipelineTriggerFilterViewModel[];
}

File: src/Core/Houston.Application/ViewModel/PipelineTriggerEventViewModels/PipelineTriggerEventViewModel.cs
using Houston.Application.ViewModel.PipelineTriggerFilterViewModels;

namespace Houston.Application.ViewModel.PipelineTriggerEventViewModels {
	public class PipelineTriggerEventViewModel {
		public Guid Id { get; set; }

		public Guid PipelineTriggerId { get; set; }

		public Guid TriggerEventId { get; set; }

		public string TriggerEvent { get; set; } = null!;

		public List<PipelineTriggerFilterViewModel> PipelineTriggerFilters { get; set; } = new List<PipelineTriggerFilterViewModel>();
	}
}
